File: 3.ödev/Colony/src/colony/AUretim.java
/**
*
* @author Sena Nur Erdem     sena.erdem1@ogr.sakarya.edu.tr
* @since 23 Mayıs 2023
* <p>
*  Rastgele bir tek sayı üreten bir üretim sınıfı
* </p>
*/
package colony;
import java.util.Random;

//AUretim sınıfı, Uretim arayüzünü uygular
class AUretim implements Uretim {
    public int uret() {
        Random rand = new Random();
        return 2 * (rand.nextInt(5)) + 1; // 1 ile 10 arası tek sayı üretir
    }
}

File: 2.ödev/ColonyClash/src/AUretim.c
/**
*
* @author Sena Nur Erdem     sena.erdem1@ogr.sakarya.edu.tr
* @since 5 Mayıs 2023
*/
#include "AUretim.h"
#include <stdlib.h>

int auretim_uret(Uretim *uretim) {
    return 2 * (rand() % 5) + 1; // 1 ile 10 arası tek sayı üretir
}

Uretim *auretim_olustur() {
    return uretim_olustur(auretim_uret);
}
